notification_endpoint.go

```diff
@@ -4,13 +4,43 @@ import (
 	"context"
 	"encoding/json"
 	"errors"
+	"time"
 )
 
 var (
 	// ErrInvalidNotificationEndpointType denotes that the provided NotificationEndpoint is not a valid type
 	ErrInvalidNotificationEndpointType = errors.New("unknown notification endpoint type")
 )
 
+// NotificationEndpointService represents a service for managing notification endpoints.
+type NotificationEndpointService interface {
+	// UserResourceMappingService must be part of all NotificationEndpointStore service,
+	// for create, delete.
+	UserResourceMappingService
```

Comment:
Why does it need this embedding?

Reply:
@desa it doesn't, the original implemntation had both orgSVC and this one embedded to use in interesting ways... mostly for testing it looked like. I just pushed up the change to nuke this dep btw, good timing 👌